perceval/backends/_mps.py

```diff
@@ -70,149 +79,230 @@ def set_circuit(self, circuit: ACircuit):
         for r, c in C:
             assert c.compute_unitary(use_symbolic=False).shape[0] <= 2, \
                 "MPS backend can not be used with components of using more than 2 modes"
-        if self._cutoff is None:
-            self._cutoff = C.m
 
     def set_input_state(self, input_state: BasicState):
         super().set_input_state(input_state)
-        self.compile()
+        self._cutoff = input_state.n + 1  # sets the default value of cut-off to max number of photons
```

Comment:
Is `input_state.n + 1` the minimum cut off value that makes MPS work?
If so, you shouldn't impose this value here, as it is overriding any value that could be set by the user, through the `__init__` or `set_cutoff` methods.

Maybe you could handle this cut off minimum value in `compile` where the maximum is already dealt with (see my other comment)

Reply:
There were some issues with setting up and clearing of previous cut_off values. Some reset (future dev) method for attributes of backends maybe good. I found some solution to not override and also perform checks on cutoff.